helix-core/src/main/java/org/apache/helix/tools/ClusterVerifiers/ZkHelixClusterVerifier.java

```diff
@@ -44,7 +44,7 @@
 import org.slf4j.LoggerFactory;
 
 public abstract class ZkHelixClusterVerifier
-    implements IZkChildListener, IZkDataListener, HelixClusterVerifier {
+    implements IZkChildListener, IZkDataListener, HelixClusterVerifier, AutoCloseable {
```

Comment:
Yes, what you said sounds correct. 
To make sure we are on the same page, `public void close()` has been implemented in this class to close zkclient and with respect to usesExternalZkClient. Its child class, e.g. BestPossibleExternalViewVerifier, there is no need to override again since it has no other resource to close. Implementing `close()` here benefits other children classes as well.

Reply:
Umm I partially agree it's just that making this base class AutoClosable is little risky. But i think it should be okay since it would "auto" close the resource only if it's implementer uses "try-with-resources" statement and this PR only modifies BestPossibleExternalViewVerifier this. One neat i can suggest here is to add clear comments about expectations from it's implementers and stating that it won't share _zkClient.